src/ts/shared/schemaType/ISOCalendarDateTimeType.ts

```diff
@@ -0,0 +1,111 @@
+import { TransformDataError } from '#shared/errors';
+import type { GenericObject } from '#shared/types';
+import { handleTypeValidation } from '#shared/utils';
+import BaseDateType from './BaseDateType';
+import type { ScalarTypeConstructorOptions } from './BaseScalarType';
+import type { SchemaTypeDefinitionBase } from './BaseSchemaType';
+import ISOCalendarDateType from './ISOCalendarDateType';
+import ISOTimeType from './ISOTimeType';
+
+export interface SchemaTypeDefinitionISOCalendarDateTime extends SchemaTypeDefinitionBase {
+	type: 'ISOCalendarDateTime';
+	dbFormat?: 's' | 'ms';
+}
+
+/** An ISOCalendarDateTime Schema Type */
+class ISOCalendarDateTimeType extends BaseDateType {
+	/** Database time format is in milliseconds */
+	private isDbInMs: boolean;
+
+	/** ISOCalendarDateType instance to use for transformations and validations of the date part of the DateTime */
+	private isoCalendarDateType: ISOCalendarDateType;
+
+	/** ISOTimeType instance to use for transformations and validations of the time part of the DateTime */
+	private isoTimeType: ISOTimeType;
+
+	public constructor(
+		definition: SchemaTypeDefinitionISOCalendarDateTime,
+		options: ScalarTypeConstructorOptions = {},
+	) {
+		super(definition, options);
+
+		const { dbFormat = 'ms' } = definition;
+		this.isDbInMs = dbFormat === 'ms';
+
+		this.isoCalendarDateType = new ISOCalendarDateType(
```

Comment:
I made these instance members rather than instantiating on the fly during validation because they required a definition to avoid type errors.  I think this is a better pattern anyway.